extensions/blocks/recurring-payments/edit.jsx

```diff
@@ -54,6 +55,16 @@ class MembershipsButtonEdit extends Component {
 			editedProductRenewInterval: '1 month',
 		};
 		this.timeout = null;
+
+		// FIXME: In Jetpack, we cannot yet support the 'new', Upgrade and Stripe Nudge-based
+		// flow, since the post-checkout redirect skips the Jetpack Onboarding Checklist in Calypso,
+		// where Akismet and VaultPress installation is triggered.
+		// Instead, we continue to show the legacy, in-block prompts to upgrade the site's plan, and to
+		// connect Stripe. This however means that we have to display the block also if the site's on a Free plan.
+		//
+		// Check p7rd6c-23C-p2 for details and progress. If ready, remove the line below, and replace all
+		// checks for `this.isJetpack` with `false`.
+		this.isJetpack = get( window, [ '_currentSiteType' ] ) !== 'simple';
```

Comment:
I've left similar notes on the PHP side of this, but this isn't quite accurate. Whether or not the nudge is used depends on a filter. That filter may enable or disable the feature on Jetpack sites at any time, now or in the future.

I'm wondering what the best way to support the filter here is 🤔 

…🤔 What if the upgrade nudge wrapper passed a prop down to the component it wraps that indicates it's inside the nudge? I think that would give us the information we need here.

Reply:
> … What if the upgrade nudge wrapper passed a prop down to the component it wraps that indicates it's inside the nudge? I think that would give us the information we need here.

That wouldn't cover the Stripe Nudge case though, would it? :thinking:

Reply:
(Fixed to use a better solution, after a call with @sirreal. Thanks for your suggestion!)